src/Stack/Build/Types.hs

```diff
@@ -280,7 +281,7 @@ data BuildOpts =
             ,boptsDryrun :: !Bool
             ,boptsGhcOptions :: ![Text]
             ,boptsFlags :: !(Map (Maybe PackageName) (Map FlagName Bool))
-            ,boptsInstallExes :: !Bool
+            ,boptsInstallExes :: !(Bool, Maybe (Path Abs Dir))
```

Comment:
I think a sum type `data InstallDest = NoInstall | DefaultInstall | InstallDir (Path Abs Dir)` would make more sense, to avoid such confusing values as `(True, Just _)`.